pkg/api/vlabs/validate.go

```diff
@@ -136,6 +137,13 @@ func (a *AgentPoolProfile) Validate(orchestratorType OrchestratorType) error {
 			return e
 		}
 	}
+	
+	//CoreOS is only applicable for Kubernetes
```

Comment:
Added the support. https://github.com/uday31in/acs-engine/commit/d64b5085a056fdf168ee2173bd75423fc9a4f2fb